common/src/main/java/com/microsoft/hydralab/performance/PerformanceTestManagementService.java

```diff
@@ -4,24 +4,40 @@
 
 import com.microsoft.hydralab.agent.runner.ITestRun;
 import com.microsoft.hydralab.agent.runner.TestRunThreadContext;
+import com.microsoft.hydralab.common.entity.common.DeviceInfo;
+import com.microsoft.hydralab.common.entity.common.TestRun;
+import com.microsoft.hydralab.common.util.FileUtil;
 import com.microsoft.hydralab.performance.inspectors.AndroidBatteryInfoInspector;
 import com.microsoft.hydralab.performance.inspectors.WindowsBatteryInspector;
 import com.microsoft.hydralab.performance.inspectors.WindowsMemoryInspector;
 import com.microsoft.hydralab.performance.parsers.AndroidBatteryInfoResultParser;
 import com.microsoft.hydralab.performance.parsers.WindowsBatteryResultParser;
 import com.microsoft.hydralab.performance.parsers.WindowsMemoryResultParser;
 import org.jetbrains.annotations.NotNull;
+import org.springframework.stereotype.Service;
 import org.springframework.util.Assert;
 
 import java.io.File;
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
 
 import static com.microsoft.hydralab.performance.PerformanceInspector.PerformanceInspectorType.*;
 import static com.microsoft.hydralab.performance.PerformanceResultParser.PerformanceResultParserType.*;
 
-public class PerformanceTestManagementService implements IPerformanceInspectionService {
+@Service
+public class PerformanceTestManagementService implements IPerformanceInspectionService, PerformanceTestListener {
+    static final ScheduledExecutorService timerExecutor = Executors.newScheduledThreadPool(5 /* corePoolSize */);
```

Comment:
Could we define this in the ThreadPoolUtil? I think we should centralize all the thread pool definitions under one class to better manage them.